mlos_bench/db/db_schema.sql

```diff
@@ -2,37 +2,137 @@
 -- Copyright (c) Microsoft Corporation.
 -- Licensed under the MIT License.
 --
+
 -- DB schema for storing MLOS benchmarking results.
 -- The syntax works for SQLite3, DuckDB, PostgreSQL, and MySQL / MariaDB.
 
 DROP TABLE IF EXISTS trial_telemetry;
-DROP TABLE IF EXISTS trial_results;
-DROP TABLE IF EXISTS trial_config;
-DROP TABLE IF EXISTS tunable_params;
+DROP TABLE IF EXISTS trial_result;
 DROP TABLE IF EXISTS experiment_merge;
 DROP TABLE IF EXISTS trial;
+DROP TABLE IF EXISTS experiment_metrics;
 DROP TABLE IF EXISTS experiment;
+DROP TABLE IF EXISTS output_metric;
+DROP TABLE IF EXISTS output_metrics;
+DROP TABLE IF EXISTS trial_config_value;
+DROP TABLE IF EXISTS trial_config;
+DROP TABLE IF EXISTS tunable_param;
+DROP TABLE IF EXISTS tunable_groups;
+
+-- A table to store the schema of tunable parameters for a given experiment.
+-- Tunable parameters and their types, used for merging the data from several experiments.
```

Comment:
Creates several indirection tables to allow better reuse of either tunable params, result sets, configs, etc. across different entities (experiments, trials, etc.).

Tries to make it easier to check whether experiment trials are mergable.

Probably still room for improvement (e.g. see `FIXME?` comments about reuse).

Reply:
Fixed those.
Open to removing the `output_metrics` related tables for now we if want.